protocol/messenger_backup_handler.go

```diff
@@ -0,0 +1,205 @@
+}
+
+func (m *Messenger) startWaitingForTheLatestBackedupMessageLoop() {
+	ticker := time.NewTicker(timeToPostponeBackedUpMessagesHandling)
+	m.backedupMessagesHandler.postponeTasksWaitGroup.Add(1)
+	go func() {
+		defer m.backedupMessagesHandler.postponeTasksWaitGroup.Done()
+		for {
+			select {
+			case <-ticker.C:
+				if !m.backedupMessagesHandler.postponeHandling {
+					return
+				}
+				m.backedupMessagesHandler.postponeHandling = false
+
+				for _, msg := range m.backedupMessagesHandler.messagesToProceed {
+					messageState := m.buildMessageState()
+					errors := m.HandleBackup(messageState, msg)
+					if len(errors) > 0 {
+						for _, err := range errors {
+							m.logger.Warn("failed to handle postponed Backup messages", zap.Error(err))
+						}
+					}
+				}
+
+				m.backedupMessagesHandler.messagesToProceed = nil
+			case <-m.quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) []error {
+	var errors []error
+	if m.backedupMessagesHandler.postponeHandling {
+		m.backedupMessagesHandler.addMessage(message)
+		return errors
+	}
+
+	err := m.handleBackedUpProfile(message.Profile, message.Clock)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	for _, contact := range message.Contacts {
+		err = m.HandleSyncInstallationContact(state, *contact)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	for _, community := range message.Communities {
+		err = m.handleSyncCommunity(state, *community)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+	err = m.handleBackedUpSettings(message.Setting)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if m.config.messengerSignalsHandler != nil {
+		response := wakusync.WakuBackedUpDataResponse{}
+
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyProfile, message.ProfileDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyContacts, message.ContactsDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyCommunities, message.CommunitiesDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeySettings, message.SettingsDetails)
+
+		m.config.messengerSignalsHandler.SendWakuFetchingBackupProgress(&response)
+	}
+
+	return errors
+}
+
+func (m *Messenger) handleBackedUpProfile(message *protobuf.BackedUpProfile, backupTime uint64) error {
+	if message == nil {
+		return nil
+	}
+
+	dbDisplayNameClock, err := m.settings.GetSettingLastSynced(settings.DisplayName)
+	if err != nil {
+		return err
+	}
+
+	response := wakusync.WakuBackedUpDataResponse{
+		Profile: &wakusync.BackedUpProfile{},
+	}
+
+	if dbDisplayNameClock < message.DisplayNameClock {
+		err := m.SetDisplayName(message.DisplayName, false)
+		response.AddDisplayName(message.DisplayName, err == nil)
+	}
+
+	syncWithBackedUpImages := false
+	dbImages, err := m.multiAccounts.GetIdentityImages(message.KeyUid)
+	if err != nil {
```

Comment:
do you know why it would fail here and why we would continue if an error is returned?

Reply:
At this point errors which may happen are only those returned from `db.Query` or `rows.Scan` call and I agree if that's the case there is nothing we can really do to recover db, but error may happen also if there are no rows in the db, for example if user hasn't set profile image.

Reply:
Ok, then we should still return an error and stop execution, unless is a `sql.ErrNoRows`, in which case we continue (there are some examples of this already in the codebase)